app/controllers/citizens/accounts_controller.rb

```diff
@@ -4,9 +4,14 @@ class AccountsController < BaseController
     before_action :authenticate_applicant!
 
     def index
-      return if worker_working?
+      return if !legal_aid_application.transactions_gathered? && worker_working?
```

Comment:
Does it mean the scenario in which the applicant adds several banks doesn't work anymore?

Reply:
instead, you could set `session[:worker_id]` in the previous controller and remove `worker_id` from the session in this controller if the job is done

```ruby
session[:worker_id] = nil unless worker_working?
```

Reply:
Or I could clear this flag in the "do you have another bank account" action

Reply:
For the record: This issues was addresses by splitting the accounts action into two - one to gather the bank transactions, and another to display them.